File: OilFieldPlatform.Calculation/OilFieldPlatform.Calculation.WebClient/src/api/schemas/requests/WaterSampleSetAutoCalcRequest.ts
export const type = 'pages.waterSample.setAutoCalc'

export interface WaterSampleSetAutoCalcRequest {
  type: 'pages.waterSample.setAutoCalc'
  isAuto: boolean
}

export function create(isAuto: boolean): WaterSampleSetAutoCalcRequest {
  return { type, isAuto }
}

File: OilFieldPlatform.Calculation/OilFieldPlatform.Calculation.Server/Schemas/Requests/WaterSampleSetAutoCalcRequest.cs
namespace OilFieldPlatform.Calculation.Server.Schemas.Requests;

/// <summary>Запрос установки режима автоматического расчёта.</summary>
public sealed class WaterSampleSetAutoCalcRequest : IWaterSamplePageRequest
{
    /// <summary>Тип сообщения.</summary>
    public string Type { get; init; } = "pages.waterSample.setAutoCalc";

    /// <summary>Флаг авто-расчёта.</summary>
    public required bool IsAuto { get; init; }
}
